etcd.py

```
@@ -426,36 +433,47 @@ def remove_unhealthy_members(self, autoscaling_members):
         for m in members.values():
             self.manager.me.delete_member(m)
 
-    @staticmethod
-    def update_record(zone, record_type, record_name, new_value):
-        records = zone.get_records()
-        old_records = [r for r in records if r.type.upper() == record_type and r.name.lower().startswith(record_name)]
-
-        if len(old_records) == 0:
-            return zone.add_record(record_type, record_name, new_value)
-
-        if set(old_records[0].resource_records) != set(new_value):
-            return zone.update_record(old_records[0], new_value)
+    def update_record(self, conn, zone_id, rtype, rname, new_value):
+        conn.change_resource_record_sets(
+            HostedZoneId=zone_id,
+            ChangeBatch={
+                'Changes': [
+                    {
+                        'Action': 'UPSERT',
+                        'ResourceRecordSet': {
+                            'Name': rname,
+                            'Type': rtype,
+                            'TTL': 60,
+                            'ResourceRecords': new_value,
+                        }
+                    }
+                ]
+            }
+        )
 
     def update_route53_records(self, autoscaling_members):
-        conn = boto.route53.connect_to_region('universal')
-        zone = conn.get_zone(self.hosted_zone)
+        conn = boto3.client('route53', region_name='universal')
+        zones = conn.list_hosted_zones_by_name(DNSName=self.hosted_zone)
+        zone = ([z for z in zones['HostedZones'] if z['Name'] == self.hosted_zone] or [None])[0]
         if not zone:
             return
```

Maybe `raise` something instead of returning silently?